.github/workflows/build.yml

```diff
@@ -0,0 +1,251 @@
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.9"
+          - "3.8"
+          - "3.7"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'corretto'
+          java-version: '11'
+      - run: make init
+      - run: pytest -vv tests/integration/workflows/java_maven
+
+  java-gradle-integration:
+    name: ${{ matrix.os }} / ${{ matrix.python }} / java gradle
+    runs-on: ${{ matrix.os }}
+    env:
+      GRADLE_OPTS: -Dorg.gradle.daemon=false
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.9"
+          - "3.8"
+          - "3.7"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - run: make init
+      - run: pytest -vv tests/integration/workflows/java_gradle
+
+  custom-make-integration:
+    name: ${{ matrix.os }} / ${{ matrix.python }} / custom make
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.9"
+          - "3.8"
+          - "3.7"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - run: make init
+      - run: pytest -vv tests/integration/workflows/custom_make
+
+  python-integration:
+    name: ${{ matrix.os }} / ${{ matrix.python }} / python
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.9"
+          - "3.8"
+          - "3.7"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - run: make init
+      - run: pytest -vv tests/integration/workflows/python_pip
+
+  ruby-integration:
+    name: ${{ matrix.os }} / ${{ matrix.python }} / ruby
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.9"
+          - "3.8"
+          - "3.7"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '^3.1'
```

Comment:
Lambda supports ruby2.7, so should we setup that version for the tests as well?

Reply:
Updated!